### ISSUE TITLE
[V3] Implement new Concept Exercise: pointers

### ISSUE THREAD

**Issue by tehsphinx** [exercism/go#1376]
This issue describes how to implement the `pointers` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to use `pointers` in Go.

## Learning objectives

- declare a pointer type
- take a pointer
- dereference a pointer
- panic on nil pointer dereference
- `new` function
- changing the underlying value of a pointer

## Out of scope

- pointer methods

## Concepts

- `pointers`

## Prerequisites

- `nil`
- `types`
- `conditionals`

## Resources to refer to

- https://dave.cheney.net/2017/04/26/understand-go-pointers-in-less-than-800-words-or-your-money-back
- https://medium.com/rungo/pointers-in-go-a789eafccd53

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by junedev** [comment]
The topics mentioned in this issues are all covered in the current Concept Exercises.
